saq/worker.py

```diff
@@ -83,6 +84,14 @@ def __init__(
 
             self.functions[name] = function
 
+    async def _before_process(self, ctx):
```

Comment:
Secretly, these are intended to be nice monkey-patching points for dd-trace-py.

Reply:
i figured as much, but why not just pass it in instead of overriding?

Reply:
ddtrace loves it's monkeypatching: https://ddtrace.readthedocs.io/en/stable/integrations.html

But that's of course pretty fragile.

I figured adding these methods was a reasonable refactor anyways, and we could make the ddtrace decision later.

Happy to remove these if we think it's ugly, though.

Reply:
Also, if we opt for passing in, we should make it possible to have multiple `before_process` and `after_process` callbacks